plutus-scb/src/Cardano/Metadata/Types.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts   #-}
+{-# LANGUAGE GADTs              #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE TemplateHaskell    #-}
+module Cardano.Metadata.Types where
+
+import           Control.Lens           (makeLenses)
+import           Control.Monad.Freer.TH (makeEffect)
+import           Data.Aeson             (FromJSON, ToJSON, toJSON)
+import qualified Data.Aeson             as JSON
+import           Data.List.NonEmpty     (NonEmpty)
+import           Data.Text              (Text)
+import           GHC.Generics           (Generic)
+import           Ledger.Crypto          (PubKey, Signature)
+import           LedgerBytes            (LedgerBytes)
+import           Servant.API            (FromHttpApiData, ToHttpApiData)
+import           Servant.Client         (BaseUrl)
+
+newtype MetadataConfig =
+    MetadataConfig
+        { mdBaseUrl :: BaseUrl
+        }
+    deriving (Show, Eq, Generic)
+    deriving anyclass (FromJSON)
+
+------------------------------------------------------------
+newtype Subject =
+    Subject Text
+    deriving (Show, Eq, Ord, Generic)
+    deriving newtype (ToJSON, FromJSON, FromHttpApiData, ToHttpApiData)
+
+newtype PropertyKey =
+    PropertyKey Text
+    deriving (Show, Eq, Ord, Generic)
+    deriving newtype (ToJSON, FromJSON, FromHttpApiData, ToHttpApiData)
+
+newtype PropertyName =
+    PropertyName Text
+    deriving (Show, Eq, Generic)
+    deriving newtype (ToJSON, FromJSON)
+
+data Property =
+    Property
+        { _propertySubject     :: Subject
+        , _propertyDescription :: PropertyDescription
+        }
+    deriving (Show, Eq, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+
+data HashFunction =
+    SHA256
+    deriving (Show, Eq, Ord, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+
+data PropertyDescription
```

Comment:
I love Haskell. This stuff is so much better than writing it out in prose :roll_eyes:

Reply:
![preach](https://media.tenor.com/images/ad5989d3f9e465fa19b877f27056c3a4/tenor.gif)